### PR TITLE
simpler plugin not variant aware

### PR DESCRIPTION
**Pull request by stephanenicolas**
The plugin now works well without being variant aware. It also offers 2 convenient methods to declare navigation api dependencies:
* `navigationApiOf(otherModuleName)` to add the navigation api of an other module
* `navigationApiOfSelf()` to add a project's own navigation api.

usage:
```groovy
client.dependencies {
  navigationApiOf(":module1")
}
producer.dependencies {
  navigationApiOfSelf()
}
```

### CODE REVIEW

**Comment by stephanenicolas** [comment]
@kshirsagarps I made the sample work with a composite build, and using the new plugin. The sample is a bit too complex, let's rework on that next week. 
There is a good documentation effort ahead, but we are getting closer to releasing DH3. :)

File: dart-sample/app/src/debug/java/com/f2prateek/dart/example/HensonNavigator.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    return Module1ActivityNavigationModel__IntentBuilder.getInitialState(context);
+  }
+  public static Module1ServiceNavigationModel__IntentBuilder.RequiredSequence<Module1ServiceNavigationModel__IntentBuilder.ResolvedAllSet> gotoModule1ServiceNavigationModel(Context context) {
+    return Module1ServiceNavigationModel__IntentBuilder.getInitialState(context);
+  }
+  public static SampleActivityNavigationModel__IntentBuilder.RequiredSequence<SampleActivityNavigationModel__IntentBuilder.ResolvedAllSet> gotoSampleActivityNavigationModel(Context context) {
+    return SampleActivityNavigationModel__IntentBuilder.getInitialState(context);
+  }
+}
+
```
Comment (dlemures) [inline dart-sample/app/src/debug/java/com/f2prateek/dart/example/HensonNavigator.java]: do we want to include the generated code?

**Comment by dlemures** [comment]
It looks good, just one question

**Comment by stephanenicolas** [comment]
I am not sure about what you mean by "include", but I cannot import any
inner class.

The code is generated in the variant source folder, I am not sure if there
is any better way to do it. I can ask Xavier.



Le 5 janv. 2018 8:02 PM, "Daniel Molinero" <notifications@github.com> a
écrit :

**Pull request merged**